### ISSUE TITLE
Support for custom types of defects

### ISSUE THREAD

**Issue by shaburov** [reportportal/reportportal#250]
![image](https://user-images.githubusercontent.com/15269131/32017667-1f98b1b4-b9d0-11e7-9c37-f7b25553076b.png)

```
Caused by: com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 400
Status message: Bad Request
Error Message: Test item status is ambiguous. Invalid test item issue type definition 'EXPECTED_IMPLEMENTATION' is requested for item '59f0dee2a7b11b0001a52fce'. Valid issue types are: [NOT_ISSUE, PRODUCT_BUG, AUTOMATION_BUG, SYSTEM_ISSUE, TO_INVESTIGATE, NO_DEFECT]
Error Type: AMBIGUOUS_TEST_ITEM_STATUS
```

**Comment by avarabyeu** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Teo](https://trello.com/c/ECl5Uawy/2-teo)

**Comment by pratik-gajjar-13988** [comment]
Is it resolved? Is there any alternative for this now?

**Comment by DzmitryHumianiuk** [comment]
@avarabyeu what is this link here for?

@pratik-gajjar-13988 @shaburov you should bypass specific ID from DB, not the name of defect in upper case.

@avarabyeu can we make WS to accept name of defect, as a parameter?
or, provide for this page, IDs of defect types, to be shown in tooltip

**Comment by shaburov** [comment]
Any option is convenient for me.

**Comment by pratik-gajjar-13988** [comment]
Thanks @dmikhievich. It really helped me.

**Comment by DzmitryHumianiuk** [comment]
@shaburov @pratik-gajjar-13988 
you can get all locators of defect types in project settings
**GET** `/api/v1/PROJECT_NAME/settings`

Actions:
* WS: output supported locators with Error message - @pbortnik 
* UI: show tooltip _onhover_ on defect type badge

**Comment by pbortnik** [comment]
@DzmitryHumianiuk , fixed, example of respnse
`{
  "error_code": 40019,
  "message": "Test item status is ambiguous. Invalid test item issue type definition 'MY_DEFECT'. Valid issue types locators are: [ND001, nd_urun1q5qqd0m, PB001, TI001, AB001, SI001]"
}`

**Comment by e-olga** [comment]
The issue is resolved.
Will be available in v 4.0
